diag_manager/fms_diag_object.F90

```diff
@@ -545,10 +547,17 @@ logical function fms_diag_accept_data (this, diag_field_id, field_data, time, is
     IF ( PRESENT(ie_in) ) ie = ie_in
     IF ( PRESENT(je_in) ) je = je_in
     IF ( PRESENT(ke_in) ) ke = ke_in
-!> Buffer the data
-    call this%FMS_diag_fields(diag_field_id)%set_data_buffer(field_data, FMS_diag_object%diag_axis,&
-                                                             is, js, ks, ie, je, ke)
+
+!> Only 1 thread allocates the output buffer amd sets set_math_needs_to_be_done
+!$omp single
+    data_buffer_is_allocated = &
+      this%FMS_diag_fields(diag_field_id)%allocate_data_buffer(field_data, FMS_diag_object%diag_axis)
```

Comment:
We need to avoid use of global variables/objects as much as we can. There could be multiple global instances of _fmsDiagObject_type_. Here, _diag_axis_ is associated to an instance _this_ but not to any other instance. Every time we change the global instance of the _fmsDiagObject_type_ we have to come back here and change accordingly which is not good. So, _FMS_diag_object%diag_axis_ should be replaced by _this%diag_axis_.